### DISCUSSION TITLE
How to expose Keda operator metrics for prometheus scrap?

### DISCUSSION THREAD

**Discussion by hueiyuan** [kedacore/keda#5834] [Q&A / Need Help]
### Description
Based on this [official document](https://keda.sh/docs/2.14/operate/prometheus/), we have tried expose 8080 port on KEDA Operator Deployment. But still can not get all of metrics about KEDA Operator and `8080/metrics` is empty.

Does have anyone know how to configure or give some example about it, thank you.

### Version
* KEDA: 2.14.2
* AWS EKS 1.29.0

**Comment by pauldotyu** [comment]
How did you install KEDA? If you installed from the Helm chart, there are flags that you need to set to true in order for the metrics to be exposed. More specifically, you'll need to set the `prometheus.operator.enabled` and `prometheus.metricServer.enabled` parameters set to true. 

Then if you want your Prometheus instanstance to scrape these metrics, you can set `prometheus.operator.serviceMonitor.enabled` and `prometheus.metricServer.serviceMonitor.enabled` to true to deploy ServiceMonitor. Just be sure to set the `prometheus.operator.serviceMonitor.additionalLabels` and `prometheus.metricServer.serviceMonitor.additionalLabels` so that the Prometheus custom resource scrapes from your smon resources.

See this doc for more:
https://github.com/kedacore/charts/blob/main/keda/README.md#operations
